templates/commands/render.go

```diff
@@ -397,21 +396,16 @@ func copyRecursive(pos *model.ConfigPos, from, to string, rfs renderFS, overwrit
 			if !os.IsNotExist(err) {
 				return model.ErrWithPos(pos, "Stat(): %w", err) //nolint:wrapcheck
 			}
+			if err := rfs.MkdirAll(dirToCreate, mkdirPerms); err != nil {
+				return model.ErrWithPos(pos, "MkdirAll(): %w", err) //nolint:wrapcheck
+			}
```

Comment:
So to confirm this is intentional - we could still create empty dirs in the destination in dryrun mode?

Reply:
That was actually a bug, thanks for catching it. I just rewrote this whole thing to make the control flow less complex and bug-prone.

Reply:
And added a test.